src/rendering/RenderUtil.cc

```diff
@@ -457,6 +520,37 @@ void RenderUtil::Update()
       visual->SetUserData("temperature", temp.second);
     }
   }
+
+  // create new collision visuals
+  {
+    if (!newCollisionLinks.empty())
+      DeselectAllEntities();
+
+    for (auto const& link : newCollisionLinks)
+    {
+      std::vector<Entity> colEntities =
+          this->dataPtr->linkToCollisionEntities[link];
+
+      for (const auto& colEntity : colEntities)
+      {
+        if (!this->dataPtr->sceneManager.HasEntity(colEntity))
+        {
+          sdf::Collision collision = this->dataPtr->entityCollisions[colEntity];
+
+          sdf::Material material;
+          material.SetAmbient(math::Color(1, 0.5088, 0.0468, 1));
+          material.SetDiffuse(math::Color(1, 0.5088, 0.0468, 1));
+          material.SetSpecular(math::Color(0.5, 0.5, 0.5, 1));
+
+          sdf::Visual visual;
+          visual.SetGeom(*collision.Geom());
+          visual.SetMaterial(material);
+          this->dataPtr->sceneManager.CreateVisual(colEntity, visual, link);
```

Comment:
The code here for creating collision visuals would be a good addition to the  `SceneManager` class, i.e. a new `CreateCollision` function

and we just have to call `rendering::VisualPtr colVisual = this->dataPtr->sceneManager.CreateCollision(colEntity, collision, link)`

Reply:
704af0dcc31636fa5f0a736752e55a1ec59bfd9a